### PR TITLE
Get signing keys from config

### PR DESCRIPTION
**Pull request by Cito**
The external public key had so far been hard coded, and the internal key pair generated on startup.

Instead, we now get both the external public key and the internal key pair from the configuration. These  configuration values should be taken from a vault, where they can be generated regularly or on demand.

If no keys are specified in the config, keys are generated automatically, so this can be run and tested without having to specify keys in the configuration. In this case, the full external key is stored as well to help with testing.

### CODE REVIEW

File: auth_service/auth_adapter/core/auth.py [hunk 1]
```diff
@@ -20,15 +20,51 @@
...
 log = logging.getLogger(__name__)
 
 
+class SigningKeys:
+    """A container for external and internal signing keys."""
+
+    external_jwks: jwk.JWKSet  # the external public key set
+    internal_jwk: jwk.JWK  # the interal key pair
+
+    # when testing, we also provide the extrnal private key
```
Comment (KerstenBreuer) [inline auth_service/auth_adapter/core/auth.py:38]:
```suggestion
    # when testing, we also provide the external private key
```

File: auth_service/auth_adapter/core/auth.py [hunk 2]
```diff
@@ -20,15 +20,51 @@
...
+    full_external_jwk: Optional[jwk.JWK] = None
```
Comment (KerstenBreuer) [inline auth_service/auth_adapter/core/auth.py:39]: Shouldn't the test related code live outside of the production code base?

File: auth_service/auth_adapter/core/auth.py [hunk 3]
```diff
@@ -20,15 +20,51 @@
...
+
+    def load(self, config: Config) -> None:
+        """Load all the signing keys from the configuration."""
+        external_keys = config.auth_ext_keys
+        if not external_keys:
```
Comment (KerstenBreuer) [inline auth_service/auth_adapter/core/auth.py:44]: What is the advantage of having this here? Maybe we can consider to generate test credentials as part of a pytest fixture.

File: auth_service/auth_adapter/core/auth.py [hunk 4]
```diff
@@ -20,15 +20,51 @@
...
+class SigningKeys:
+    """A container for external and internal signing keys."""
+
+    external_jwks: jwk.JWKSet  # the external public key set
+    internal_jwk: jwk.JWK  # the interal key pair
+
+    # when testing, we also provide the extrnal private key
+    full_external_jwk: Optional[jwk.JWK] = None
+
+    def load(self, config: Config) -> None:
```
Comment (KerstenBreuer) [inline auth_service/auth_adapter/core/auth.py:41]: Can we make this the `__init__` method?

File: auth_service/auth_adapter/core/auth.py [hunk 5]
```diff
@@ -20,15 +20,51 @@
...
+__all__ = ["exchange_token", "signing_keys"]
 
 log = logging.getLogger(__name__)
 
 
+class SigningKeys:
+    """A container for external and internal signing keys."""
+
+    external_jwks: jwk.JWKSet  # the external public key set
+    internal_jwk: jwk.JWK  # the interal key pair
```
Comment (KerstenBreuer) [inline auth_service/auth_adapter/core/auth.py:36]: These are not defined upon `__init__`.

File: auth_service/config.py [hunk 6]
```diff
@@ -62,6 +62,8 @@ class Config(ApiConfigBase):
     service_name: str = "auth_service"
     log_level: LogLevel = "debug"
     run_auth_adapter: bool = False
+    auth_int_keys: Optional[str] = None
```
Comment (KerstenBreuer) [inline auth_service/config.py:65]: Is this a pair of secret and public key? If so, shouldn't we inject as separate parameters? Would be much easier to store them separately in vault.

**Review by KerstenBreuer — APPROVED** [review]
Maybe we can achieve a nice separation between test and production logic. Otherwise nice.

File: auth_service/auth_adapter/core/auth.py (hunk 3 above)
Comment (Cito) [reply auth_service/auth_adapter/core/auth.py:44]: Will be refactored as discussed.

File: auth_service/config.py (hunk 6 above)
Comment (Cito) [reply auth_service/config.py:65]: They are stored as a JWK structure, so it's easier to have them merged.

File: auth_service/auth_adapter/core/auth.py (hunk 5 above)
Comment (Cito) [reply auth_service/auth_adapter/core/auth.py:36]: This has now been refactored.

File: auth_service/auth_adapter/core/auth.py (hunk 2 above)
Comment (Cito) [reply auth_service/auth_adapter/core/auth.py:39]: Has been changed now as discussed.

**Pull request merged**